### PR TITLE
Unite the Language feature checking functions into it's own namespace. 

### PR DESCRIPTION
**Pull request by AdamSpeight2008**
Issue #13228
Bring them under the namespace `Microsoft.CodeAnalysis.VisualBasic.LanguageFeatures`.
So that the checks are implemented via a singe route, 
These methods are available as extension methods, on `Feature` enum, 
```vbnet
If Feature.InterpolatedString.IsAvailable( thisCompilation ) Then
 ',,,
End If
```
or on the `InternalSyntax.VisualBasicSyntaxNode`, which if required can attach a diagnostic to it.

The **Required Language Version** are now provided via a singleton, allowing the sharing of instances between diagnostics. Small reduction in allocations in code that uses a feature unavailable in a particular version of the language.
*Note: There is a small use of reflection, to initially setup the dictionary and the instances.*

### CODE REVIEW

File: src/Compilers/VisualBasic/Portable/LanguageFeatures/LanguageFeature.vb [hunk 1]
```diff
@@ -39,6 +39,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
...
     Friend Module FeatureExtensions
+        <Extension>
+        Friend Function GetFeatureFlag(feature As Feature) As String
+            Select Case feature
+                'Case feature.IOperation
+                '    Return "IOperation"
+
+                Case Else
+                    Return Nothing
+            End Select
```
Comment (jaredpar) [inline src/Compilers/VisualBasic/Portable/LanguageFeatures/LanguageFeature.vb]: This method is incomplete

File: src/Compilers/VisualBasic/Portable/LanguageFeatures/LanguageFeatureRequiredVersionService.vb [hunk 2]
```diff
@@ -0,0 +1,191 @@
+﻿Imports Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
```
Comment (jaredpar) [inline src/Compilers/VisualBasic/Portable/LanguageFeatures/LanguageFeatureRequiredVersionService.vb]: This file needs a copyright header.

File: src/Compilers/VisualBasic/Portable/Parser/ParseStatement.vb [hunk 3]
```diff
@@ -5,6 +5,8 @@
 '-----------------------------------------------------------------------------
 Imports Microsoft.CodeAnalysis.Syntax.InternalSyntax
 Imports InternalSyntaxFactory = Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.SyntaxFactory
+Imports Microsoft.CodeAnalysis.VisualBasic.LanguageFeatures.CheckFeatureAvailability
+
```
Comment (jaredpar) [inline src/Compilers/VisualBasic/Portable/Parser/ParseStatement.vb]: Extra blank line

File: src/Compilers/VisualBasic/Portable/Parser/Parser.vb [hunk 4]
```diff
@@ -74,6 +75,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             End Get
         End Property
 
+        Public ReadOnly Property Options() As VisualBasicParseOptions
```
Comment (jaredpar) [inline src/Compilers/VisualBasic/Portable/Parser/Parser.vb]: No need for parens on the Options name.

File: src/Compilers/VisualBasic/Portable/LanguageFeatures/LanguageFeatureRequiredVersionService.vb [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+        Friend Sub New(version As LanguageVersion)
+            Me.Version = version
+        End Sub
+
+        Public Overrides Function ToString() As String
+            Return Version.ToDisplayString()
+        End Function
+    End Class
+
+    Module CheckFeatureAvailability
```
Comment (jaredpar) [inline src/Compilers/VisualBasic/Portable/LanguageFeatures/LanguageFeatureRequiredVersionService.vb]: Need an accessibility modifier

Comment (jaredpar) [reply src/Compilers/VisualBasic/Portable/LanguageFeatures/LanguageFeatureRequiredVersionService.vb]: I would move this module into a separate .vb file and also remove the `#region` blocks. Class doesn't seem to warrant them.

File: src/Compilers/VisualBasic/Portable/LanguageFeatures/LanguageFeature.vb (hunk 1 above)
Comment (AdamSpeight2008) [reply src/Compilers/VisualBasic/Portable/LanguageFeatures/LanguageFeature.vb]: Are you sure? From my understand it returns `Nothing` for all cases, when the code is commented out. 
Which is do to `IOperation` not being in the definitive feature list.

Comment (jaredpar) [reply src/Compilers/VisualBasic/Portable/LanguageFeatures/LanguageFeature.vb]: What method is this replacing? The only use of IOperation is in the new code here and it is commented out.

Comment (AdamSpeight2008) [reply src/Compilers/VisualBasic/Portable/LanguageFeatures/LanguageFeature.vb]: It is for features that are enabled via the cmd line feature flag.

**Comment by AdamSpeight2008** [comment]
Test failure due to PR#25083 not being present.

**Comment by AdamSpeight2008** [comment]
@dotnet-bot Retest this please.

**Comment by AdamSpeight2008** [comment]
@dotnet-bot test this please

**Comment by AdamSpeight2008** [comment]
@dotnet-bot test this please

**Comment by AdamSpeight2008** [comment]
@dotnet-bot retest this please

**Pull request closed without merging**